data/src/main/java/com/android254/data/repos/AuthManager.kt

```diff
@@ -23,23 +23,28 @@ import com.android254.data.network.util.TokenProvider
 import com.android254.domain.models.DataResult
 import com.android254.domain.models.Success
 import com.android254.domain.repos.AuthRepo
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
 import javax.inject.Inject
 
 class AuthManager @Inject constructor(
     private val api: AuthApi,
     private val tokenProvider: TokenProvider
 ) : AuthRepo {
     override suspend fun getAndSaveApiToken(idToken: String): DataResult<Success> {
-        return try {
-            val result = api.googleLogin(GoogleToken(idToken))
-            tokenProvider.update(result.token)
-            DataResult.Success(Success)
-        } catch (e: Exception) {
-            when (e) {
-                is ServerError, is NetworkError -> {
-                    DataResult.Error("Login failed", networkError = true, exc = e)
-                } else -> {
-                    DataResult.Error("Login failed", exc = e)
+        return withContext(Dispatchers.IO) {
```

Comment:
It is recommended you inject dispatchers using DI instead of hardcoded ones: Read here:[ ](https://developer.android.com/kotlin/coroutines/coroutines-best-practices#inject-dispatchers)